source/_posts/en/tutorial/background.md

```diff
@@ -1,71 +1,93 @@
+
+It provides the best choice for [Ryan Dahl](http://tinyclouds.org/), who want to write a high-performance server and with no progress for a long time.
+
+Soon after, at the JSConf in Berlin in 2009, JavaScript-based server project Node.js officially released.
+
+It gives JavaScript developer a high-performance server and highly promotes workflow tools for Front-End development.
+
+In the meantime, because of the performance improved, more and more business logic migrate from server side to browser.
+
+Browser-side business logic becomes complex, and developer realized that they need a framework to solve these problems.
+
+San, the MVVM JavaScript framework. The architecture solution for web development. 
+
+Concept
 -----
 
-MVVM 模式，顾名思义即 Model-View-ViewModel 模式。它萌芽于2005年微软推出的基于 Windows 的用户界面框架 WPF ，前端最早的 MVVM 框架 [knockout](https://github.com/knockout/knockout) 在2010年发布。
+MVVM(Model-View-ViewModel) pattern. First invented in WPF framework from windows user application framework in 2005. 
 
-一句话总结 Web 前端 MVVM：操作数据，就是操作视图，就是操作 DOM（所以无须操作 DOM ）。
+The earliest MVVM JavaScript framework -- [knockout](https://github.com/knockout/knockout), was released in 2010. 
 
-无须操作 DOM ！借助 MVVM 框架，开发者只需完成包含 **声明绑定** 的视图模板，编写 ViewModel 中业务数据变更逻辑，View 层则完全实现了自动化。这将极大的降低前端应用的操作复杂度、极大提升应用的开发效率。MVVM 最标志性的特性就是 **数据绑定** ，MVVM 的核心理念就是通过 **声明式的数据绑定** 来实现 View 层和其他层的分离。完全解耦 View 层这种理念，也使得 Web 前端的单元测试用例编写变得更容易。
+With MVVM pattern, data are equal to views, and there is no need to write any code to operate the DOM. All the DOM mutations are take over by MVVM frameworks. 
 
-MVVM，说到底还是一种分层架构。它的分层如下：
+JavaScript developer only need to write template including **declare bindings syntax**, the business logic focused on application data in ViewModel. 
 
-- Model: 域模型，用于持久化
-- View: 作为视图模板存在
-- ViewModel: 作为视图的模型，为视图服务
+The template will update automatically when the binding data changed. This design deeply reduces the complexity of the application and increase the efficiency of application development.
 
-### Model 层
+The most significant feature of MVVM pattern is the **data bindings**. It use **data bindings** syntax to separate the View layer from other layers and without any UI code. Unit test code becomes more easily.
 
-Model 层，对应数据层的域模型，它主要做`域模型的同步`。通过 Ajax/fetch 等 API 完成客户端和服务端业务 Model 的同步。在层间关系里，它主要用于抽象出 ViewModel 中视图的 Model。
+After all, MVVM pattern is a Hierarchical architecture, organized as followed: 
 
-### View 层
+- Model: used to store persistence data
+- View: the template including **data bindings** syntax
+- ViewModel: the mirror of template in data ways, auto sync changes with view. 
 
-View 层，作为视图模板存在，在 MVVM 里，整个 View 是一个动态模板。除了定义结构、布局外，它展示的是 ViewModel 层的数据和状态。View 层不负责处理状态，View 层做的是 **数据绑定的声明**、 **指令的声明**、 **事件绑定的声明**。
+### Model
 
-### ViewModel 层
+Modal layer. The model used to store data. Use Ajax/fetch API to keep client and server data sync. It used to abstract the View's Model of ViewModel in layer relations. 
```

Comment:
Modal layer. The model used to store data. &rarr; The model layer is used to store data.

Reply:
We can sync data between client server with AJAX/fetch API.